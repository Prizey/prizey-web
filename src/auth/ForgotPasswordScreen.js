import React from 'react'
import { Redirect } from '@reach/router'
import { ForgotPassword } from 'croods-auth'
import { ForgotPassword as ForgotPasswordForm } from 'seasoned-auth-forms-web'

import AuthForm from './AuthForm'

export const appendRedirect = create => values =>
  create({
    ...values,
    redirectUrl: `${process.env.REACT_APP_WEB_URL}/reset-password`,
  })

export default props => (
  <ForgotPassword
    {...props}
    render={renderProps =>
      AuthForm(ForgotPasswordForm)({
        ...renderProps,
        create: appendRedirect(renderProps.create),
      })
    }
    renderCreated={() => <Redirect to="/forgot-password/sent" noThrow />}
  />
)
