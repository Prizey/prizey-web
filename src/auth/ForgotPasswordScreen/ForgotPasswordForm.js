import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { ForgotPassword } from 'seasoned-auth-forms-web'

import SubmitButton from 'design/SubmitButton/SubmitButton'
import SecondaryLink from 'design/SecondaryLink'

const styles = theme => ({
  links: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: theme.spacing.sm,
  },
})

export const FormBottom = withStyles(styles)(({ classes, ...props }) => (
  <React.Fragment>
    <SubmitButton {...props} label="SEND EMAIL" />
    <div className={classes.links}>
      <SecondaryLink to="/sign-in" label="Already have an account?" />
    </div>
  </React.Fragment>
))

export default props => (
  <ForgotPassword
    {...props}
    renderButton={buttonProps => <FormBottom {...buttonProps} />}
  />
)
