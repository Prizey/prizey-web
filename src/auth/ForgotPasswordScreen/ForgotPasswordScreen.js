import React from 'react'
import { Redirect } from '@reach/router'
import { ForgotPassword } from 'croods-auth'
import ForgotPasswordForm from './ForgotPasswordForm'
import createAuthForm from '../AuthForm'

const AuthForm = createAuthForm(ForgotPasswordForm, 'Reset Your Password')

export const appendRedirect = create => values =>
  create({
    ...values,
    redirectUrl: `${process.env.REACT_APP_WEB_URL}/reset-password`,
  })

export default props => (
  <ForgotPassword
    {...props}
    render={renderProps => (
      <AuthForm {...renderProps} create={appendRedirect(renderProps.create)} />
    )}
    renderCreated={() => <Redirect to="/forgot-password/sent" noThrow />}
  />
)
