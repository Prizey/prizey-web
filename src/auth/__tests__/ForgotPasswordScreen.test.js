import React from 'react'
import renderer from 'react-test-renderer'

import ForgotPasswordScreen, {
  appendRedirect,
} from '../ForgotPasswordScreen/ForgotPasswordScreen'

jest.mock('croods-auth', () => ({
  ForgotPassword: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ user: { id: 1 } })}</div>
    </div>
  ),
}))

jest.mock('seasoned-auth-forms-web', () => ({
  ForgotPassword: props => (
    <div {...props}>ForgotPassword - {props.renderButton(props)}</div>
  ),
}))

jest.mock('@reach/router', () => ({
  Link: props => <div {...props}>Link - {props.children}</div>,
  Redirect: props => <div {...props}>Redirect - {props.children}</div>,
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ id: 1 })}</div>
    </div>
  ),
}))

jest.mock('../../design/Layout/RegisterPageView')

it('renders correctly', () => {
  const tree = renderer.create(<ForgotPasswordScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('apppend a token on create', () => {
  const data = {
    email: 'my@email.com',
  }
  const create = jest.fn()

  appendRedirect(create)(data)
  expect(create).toHaveBeenCalledWith({
    ...data,
    redirectUrl: `${process.env.REACT_APP_WEB_URL}/reset-password`,
  })
})
