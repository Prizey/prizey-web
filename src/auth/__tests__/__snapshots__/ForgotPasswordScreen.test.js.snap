// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    render - 
    <div
      className="Component-root-9"
    >
      <div
        className="Component-header-7"
      />
      <div
        className="Component-container-4"
      >
        <img
          alt="Prizey"
          className="Component-logoImage-8"
          src="/logo.png"
        />
        <div
          className="Component-caption-2"
        />
        <div
          className="Component-children-3"
        >
          <h5
            className="MuiTypography-root-10 MuiTypography-h5-26 MuiTypography-alignCenter-33 Component-pageTitle-1"
          >
            Reset Your Password
          </h5>
          <div
            onSubmit={[Function]}
            renderButton={[Function]}
          >
            ForgotPassword - 
            <button
              className="MuiButtonBase-root-72 MuiButton-root-46 MuiButton-contained-57 MuiButton-containedPrimary-58 MuiButton-raised-60 MuiButton-raisedPrimary-61 MuiButton-fullWidth-71"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onSubmit={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              renderButton={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-47"
              >
                SEND EMAIL
              </span>
              <span
                className="MuiTouchRipple-root-75"
              />
            </button>
          </div>
        </div>
        <div
          className="Component-footer-5"
        >
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              TERMS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              PRIVACY
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              FAIRNESS
            </span>
          </a>
          <a
            className="Component-footerLink-6"
            href="https://termly.io/en/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-10 MuiTypography-caption-20 MuiTypography-colorTextSecondary-43 MuiTypography-alignCenter-33"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
    <div
      noThrow={true}
      to="/forgot-password/sent"
    >
      Redirect - 
    </div>
  </div>
</div>
`;
